frb_dart/lib/src/ffi/io.dart

```diff
@@ -42,35 +29,177 @@ class DartApiDl {
   void initApi() {
     _initCode ??= _initFn(ffi.NativeApi.initializeApiDLData);
     if (_initCode != 0) {
-      throw 'Failed to initialize Dart API. Code: $_initCode';
+      throw Exception('Failed to initialize Dart API. Code: $_initCode');
     }
   }
 }
 
-// NOTE for maintainer: Please manually keep in sync with [WireSyncReturnStruct] in Rust
-/// This class is only for internal usage.
-class WireSyncReturnStruct extends ffi.Struct {
-  /// Not to be used by normal users, but has to be public for generated code
-  external ffi.Pointer<ffi.Uint8> ptr;
+typedef WireSyncReturn = ffi.Pointer<DartCObject>;
+
+dynamic wireSyncReturnIntoDart(WireSyncReturn syncReturn) =>
+    syncReturn.ref.intoDart();
 
-  /// Not to be used by normal users, but has to be public for generated code
-  @ffi.Int32()
-  external int len;
+extension DartCObjectExt on DartCObject {
```

Comment:
uber-nit: About the code that converts DartCObject into dart data - shall we extract it into a separate file, so it is clearer